pyOCD/tools/gdb_server.py

```diff
@@ -62,9 +66,9 @@ def build_parser(self):
         parser.add_argument("--list-targets", action="store_true", dest="list_targets", default=False, help="List all available targets.")
         parser.add_argument("--json", action="store_true", dest="output_json", default=False, help="Output lists in JSON format. Only applies to --list and --list-targets.")
         parser.add_argument("-d", "--debug", dest="debug_level", choices=debug_levels, default='info', help="Set the level of system logging output. Supported choices are: " + ", ".join(debug_levels), metavar="LEVEL")
-        parser.add_argument("-t", "--target", dest="target_override", choices=supported_targets, default=None, help="Override target to debug.  Supported targets are: " + ", ".join(supported_targets), metavar="TARGET")
```

Comment:
If I use a 'type' function that raises `argparse.ArgumentError` then the error message becomes:
```
usage: pyocd-gdbserver [-h] [--version] [-p PORT_NUMBER] [-T TELNET_PORT]
                       [--allow-remote] [-b BOARD_ID] [-l] [--list-targets]
                       [--json] [-d LEVEL] [-t TARGET] [-n] [-r]
                       [-C VECTOR_CATCH] [-s] [-f FREQUENCY] [-o] [-bh]
                       [-ce | -se] [-u] [-hp] [-fp] [-S] [-G]
                       [-c CMD [CMD ...]] [-da DAPARG [DAPARG ...]]
pyocd-gdbserver: error: argument -t/--target: invalid target value: 'foo'
```

instead of:
```
Error: unsupported target 'foobar' specified

Available targets: kl25z, kl26z, lpc824, k82f25615, lpc11xx_32,
kinetis, lpc800, lpc4088qsb, maxwsnenv, kl05z, k64f, lpc1768, lpc4088,
lpc4330, max32600mbed, k66f18, w7500, ke18f16, k22f, lpc4088dm,
ke15z7, kv11z7, nrf51, nrf52, kv10z7, k20d50m, kl46z, stm32f103rc,
kl27z4, kw40z4, cortex_m, lpc11u24, stm32f051, kl02z, ncs36510, kl28z,
kl43z4, kw01z4
```

I like the usage being shown in the former, and having the same error message that you'd see from other invalid parameters. Otoh, it's nice to see a list of valid targets. Doesn't seem I can change the error message if I let argparse catch the exception (and the type function has to be called "target" for the message to be "invalid target value"). Alternatively, I could raise another exception class that is not caught by argparse, and handle it at the top level.

Which error style do you prefer?